docs: Add a Testmonials page. (#3005)
martinvonz/jj: docs/testimonials.md

practice. In order to provide some motivation, we've collected a number of real,
100% authentic testimonials &mdash; from our loving users, our silly developers
&mdash; all to tip the scales and get you on our side!

## What the users have to say 

> I've spent many years of my career working on version control. What I like
> most about Jujutsu is how it has non-obvious solutions to UX problems that
> we've run into in the past. What most people may not realize is that there are
> many novel features which all interlock to make it easy to use.
>
> For example, consider Jujutsu's support for automatically rebasing descendants
> of amended revisions. When we implemented that in Mercurial, we ran into an
> issue: what if there's a merge conflict? Our solution was to warn users and
> just not perform the auto-rebase. Now, suddenly, users have to understand that
> there can be old versions of the same revision visible in their log, and learn
> how to fix this state.
>
> In contrast, Jujutsu's solution is to simply make merge conflicts first-class.
> This is not just an improvement in general, it is also specifically an

> > improvement for auto-rebase &mdash; users no longer have to learn about old

Reviewer: Interestingly, most style guides say that you [shouldn't put spaces around em-dashes](https://prowritingaid.com/grammar/1000066/Should-I-use-spaces-around-an-em-dash). Like many people I've been in the habit of putting spaces around em-dashes, though I've been trying to move away from that.

Either way's fine and consistency is most important, but just wanted to mention that.

&mdash; Ben, who doesn't want you keeping your existing workflow

> Wait, it's not called Jujitsu?

&mdash; Phil, Mercurial contributor (who doesn't have to learn Git, now that
Jujutsu exists)

## What the developers have to say

> I've been a FOSS contributor using Git for over 16 years, and Jujutsu
> continues to amaze me every day. It has that sweet simplicity I was fond of in
> Darcs, but it boils down all my most core and fundamental workflows &mdash;
> developed over years of experience &mdash; into a simple set of primitives.
> The internal design is simple and beautiful; it looks like a database, making
> the implementation elegant, safe, and extensible. All this, using the same Git
> repositories my coworkers use.
>
> It's like if you found out one day that you built your entire home on a vein
> of rich gold. Every day I seem to find new and beautiful emergent behaviors,
> all adding up to a tool that is greater than the sum of its parts.

&mdash; Austin Seipp, "No 1. Jujutsu Fan"

> Honestly, I implemented signing support mostly for that sweet dopamine hit
> that you get from the green checkmark on GitHub. Yeah.

&mdash; Anton Bulakh, contributor and dopamine enthusiast

> > I'm sometimes still surprised that navigating with `next` and `prev` works.

Reviewer 2: ```suggestion
> I'm sometimes still surprised that navigating with `jj next` and `jj prev` works.
```

should probably be symmetric with the punchline

Author: done

# Testimonials

## Read what users and developers have to say: 

> I'm surprised when it works

Martin von Zweigbergk, project lead who contributed over 100k lines of code.

> Jujutsu is pretty cool, you can even keep most of your existing workflows 

Ben, who really doesn't want you keeping your existing workflow.

> I'm sometimes still surprised that the navigation works.

Philip, initial contributor of `next` and `prev`. 

> Who are you? What are you doing at my house?  

Austin Seipp, contributor.

> Wait, it's not called Jujitsu?

> Phil, an Mercurial contributor and user.

Reviewer 3: an -> a

Author: done.

> 
> Over and over, I'm struck by how well Jujutsu demonstrates this kind of 
> evolved thinking, As an experienced version control developer, 
> I deeply appreciate it.

Rain, engineer at Oxide and former VCS developer

> Jujutsu is pretty cool, you can even keep most of your existing workflows 

Ben, who really doesn't want you keeping your existing workflow.

> Wait, it's not called Jujitsu?

Phil, a Mercurial contributor and user.

> It's the easiest time I've ever had learning a tool this deeply this quickly, 
> because of the ability to experiment and undo, instead of triple-checking 
> before trying a new scary command.

Scott Olson, a pretty advanced Git user who replaced it completely with 
Jujutsu.

> Jujutsu is amazing because it trivializes some of the things I couldn't have come 
> up with anything remotely as elegant. 
> It's so rare that a solution attacks the innermost core of a problem so 
> thoroughly, I genuinely feel blessed to be in its presence. And also a bit 
> vindicated in not even trying to learn to use any of the tools that felt like
> more crutches stacked upon a sand castle  

An anonymous user.

> only breaks half as often as my other tools!

> jyn, ex-rustc bootstrap developer, a place where tools break down all the time.

Author: @jyn514, is this good?

## What the developers have to say 

> I'm surprised when it works

Martin von Zweigbergk, project lead who contributed over 100k lines of code.

> I'm sometimes still surprised that navigating with `next` and `prev` works.

Philip Metzger, contributor, author of `jj next` and `jj prev`. 

> The phone number you have dialed is no longer in service. Please hang up and try a new number.

Austin Seipp, contributor.

> > As a fan of big green checkmarks in the log, made it so that Github is a ppure dopamine hit. 

Reviewer 2: ```suggestion
> As a fan of big green checkmarks in the log, jj made it so that Github is a pure dopamine hit. 
```

Author: That's a great suggestion, thanks. I've only replace `jj` with Jujutsu.

> Anton Bulakh, contributor, author of the cryptographic commit signing support.

Author: Does that look good @necauqua?

## Spread the word yourself

<!-- TODO: Reword the sentence mentioning a PR team, as it does not look good with a Google associaton -->
Satisfied? Overwhelmed with joy? Ready to recommend Jujutsu to a Jujillion 
people? Good, because we have list of Pre-Approved Endorsements, designed in a 
laboratory by our PR team, ready to go. Start shilling for us immediately and 
your check will be in the mail soon enough!

> Jujutsu seems alright.

> Jujutsu is my favorite software tool of all time. I say this for no particular reason.

> > I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. I love Jujutsu. 

Reviewer 2: Big fan of the "GPT hallucination vibes" with this one

## What the users have to say 

> I've spent many years of my career working on version control. What I like most
> about Jujutsu is how it has non-obvious solutions to UX problems that we've run
> into in the past. Most people may not realize this at first, but the many novel
> features in Jujutsu all interlock to make it easy to use.
> 
> For example, consider Jujutsu's support for automatically rebasing descendants 
> of amended revisions. When we implemented that in Mercurial, we ran into an 
> issue: what if there's a merge conflict? Our solution was to warn users if that
> occurred, and just not perform the auto-rebase. Now, suddenly, users have to 
> understand that there can be old versions of the same revision visible in their
> log. They also need to learn how to get themselves out of this state.
> 
> In contrast, Jujutsu's solution is to simply make merge conflicts first-class.
> This is not just an improvement in general, it is also specifically an
> improvement for auto-rebase—users no longer have to learn about old 
> versions of a revision unless they want to look at the obslog.
> 
> Over and over, I'm struck by how well Jujutsu demonstrates this kind of 

> > evolved thinking, As an experienced version control developer, 

Reviewer: ```suggestion
> evolved thinking. As an experienced version control developer, 
```

Reviewer: (not your fault, I just caught it myself)

Ben, who really doesn't want you keeping your existing workflow.

> Wait, it's not called Jujitsu?

Phil, a Mercurial contributor and user.

> It's the easiest time I've ever had learning a tool this deeply this quickly, 
> because of the ability to experiment and undo, instead of triple-checking 
> before trying a new scary command.

Scott Olson, a pretty advanced Git user who replaced it completely with 
Jujutsu.

> Jujutsu is amazing because it trivializes some of the things I couldn't have come 
> up with anything remotely as elegant. 
> It's so rare that a solution attacks the innermost core of a problem so 
> thoroughly, I genuinely feel blessed to be in its presence. And also a bit 
> vindicated in not even trying to learn to use any of the tools that felt like
> more crutches stacked upon a sand castle  

An anonymous user.

> only breaks half as often as my other tools!

jyn, ex-rustc bootstrap developer, a place where tools break down all the time.

> I initially started to use Jujutsu for personal repos where it has quickly 
> gone from "neat, let's try this more" to "very neat, added to my permanent
> config and automatically installed for new machines". 

> Poliorcetics

Author: @poliorcetics, thanks for the testimonial, is this good for you?

Reviewer 4: English question (not a native speaker): is it better to say `personal repos where it has` or `personal repos and it has` ? I'm unsure about the differences and I used `where` by habit I think

Reviewer 5: i would either say "personal repos, where" or "personal repos and". "personal repos where" without a comma looks odd to my eye.

Reviewer: I think either gets the point across, though I would personally use `and it has`, and also add a comma:

`Jujutsu for personal repos, and it has`

But that's just my stylistic choice. It's completely fine as-is.

Reviewer 4: Then @PhilipMetzger `, and` or `, where` are fine :)

Author: I've added the comma (`, and`) and reverted it to the fully spelled out version.

Reviewer 4: Yes thanks :)

Author: No, thank you for your contribution :-).
